mono/metadata/boehm-gc.c

```diff
@@ -81,7 +81,11 @@ typedef struct {
 	guint32  *bitmap;
 	gpointer *entries;
 	guint32   size;
+#ifdef __cplusplus
```

Comment:
This is not required -- could cast elsewhere instead. If there were a public type I'd err toward no change so it is guaranteed the same layout C vs. C++. While it is 8 bits either way, I wouldn't be sure about padding on either side being the same.